### ISSUE TITLE
GPU usage percentage of processes always shows 0%.

### ISSUE THREAD

**Issue by angel-devicente** [Syllo/nvtop#177]
Hello,

with nvtop version 3.0.1 in ArchLinux the GPU usage percentage of each process seems to always show 0%. I attach an image.

![Screenshot_2022-11-16_11-11-49](https://user-images.githubusercontent.com/10155829/202167574-81bcca34-f5c6-456d-9795-6a1ee8b94881.png)

**Comment by airvzxf** [comment]
The same is happening to me. It looks like the [issue #110](https://github.com/Syllo/nvtop/issues/110) is not resolving the problem.

![Screenshot 2023-04-12_19-25-13](https://user-images.githubusercontent.com/831380/231621731-72f6e599-dafb-428a-b500-69227ac1b38e.png)

I tested [nvitop](https://github.com/XuehaiPan/nvitop), and it has the same issue. This image shows that the GPU is using 13% (UTL), but the processes are using 1% (%SM column) of it. Furthermore, it is a discrepancy between the GPU UTL between the bar graphic on the top-right and the value for the same subject on the middle-right: 13% versus 9%.

![Screenshot 2023-04-12_19-31-38](https://user-images.githubusercontent.com/831380/231624172-1be51754-ffc4-4dbd-a5b4-35c6eba9009f.png)

```

                   -`
                  .o+`                 wolf@MSIGT73EVR7RF
                 `ooo/                 OS: Arch Linux
                `+oooo:                Kernel: x86_64 Linux 6.2.10-arch1-1
               `+oooooo:               Uptime: 2d 1h 49m
               -+oooooo+:              Packages: 1720
             `/:-:++oooo+:             Shell: zsh 5.9
            `/++++/+++++++:            Resolution: 1920x1080
           `/++++++++++++++:           WM: OpenBox
          `/+++ooooooooooooo/`         WM Theme: Clearlooks
         ./ooosssso++osssssso+`        GTK Theme: Adwaita [GTK3]
        .oossssso-````/ossssss+`       Disk: 296G / 469G (67%)
       -osssssso.      :ssssssso.      CPU: Intel Core i7-7820HK @ 8x 3.9GHz [57.0°C]
      :osssssss/        osssso+++.     GPU: NVIDIA GeForce GTX 1080
     /ossssssss/        +ssssooo/-     RAM: 11893MiB / 64265MiB
   `/ossssso+/:-        -:/+osssso+-
  `+sso+:-`                 `.-/+oso:
 `++:.                           `-/+/
 .`                                 `/
```

It could be something wrong with Nvidia for Linux. I have a dual boot, in this Laptop, then on Windows it shows the GPU Fan speed but in Linux not.

![Screenshot 2023-04-12_19-55-04](https://user-images.githubusercontent.com/831380/231626689-2978602c-9d1c-4a94-a979-acf5712abbce.png)

**Comment by qwertychouskie** [comment]
Considering that this issue applies to another unrelated monitoring app, it sounds like an Nvidia driver issue rather than an `nvtop` issue.  Should probably close as notourbug, though maybe also add a note somewhere that this feature may be broken.

**Comment by airvzxf** [comment]
I agree, it should be the NVIDIA driver which not returning the correct data. The best approach that I found to check the usage of the GPU is to measure the power, the energy consumes.

![Screenshot_2023-12-15_08-19-58](https://github.com/Syllo/nvtop/assets/831380/8c095940-f71e-4986-aa73-e27248459efb)

**Comment by Syllo** [comment]
I'll update the NVML support to call the newest API function versions. It may help with this problem and with GPU MIG/virtualization modes.
